hathor/indexes/rocksdb_utxo_index.py

```diff
@@ -0,0 +1,412 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import struct
+from dataclasses import dataclass
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Iterator, NewType, Optional
+
+from structlog import get_logger
+
+from hathor.conf import HathorSettings
+from hathor.crypto.util import decode_address, get_address_b58_from_bytes
+from hathor.indexes.rocksdb_utils import RocksDBIndexUtils
+from hathor.indexes.utxo_index import UtxoIndex, UtxoIndexItem
+
+if TYPE_CHECKING:  # pragma: no cover
+    import rocksdb
+
+settings = HathorSettings()
+logger = get_logger()
+
+_DB_NAME: str = 'utxos'
+_CF_NAME_UTXO_INDEX = b'utxo-index'
+
+# the following type is used to help a little bit to distinguish when we're using a byte sequence that should only be
+# internally used
+_InternalUid = NewType('_InternalUid', bytes)
+_INTERNAL_HATHOR_TOKEN_UID = _InternalUid(b'\x00' * 32)
+
+
+def _to_internal_token_uid(token_uid: bytes) -> _InternalUid:
```

Comment:
We already have this in other index, right? Should we try to reuse it?

Reply:
Makes sense, since this could be recurring in rocksdb indexes implementations I'll move it to `hathor.index.rocksdb_utils`